op_builder/cpu/comm.py

```diff
@@ -46,6 +47,7 @@ def extra_ldflags(self):
 class ShareMemCommBuilder(CPUOpBuilder):
     BUILD_VAR = "DS_BUILD_SHM_COMM"
     NAME = "deepspeed_shm_comm"
+    machine = platform.machine().lower()
```

Comment:
`machine` should be a member of `CPUOpBuilder`.  Suggest to define `machine` in `CPUOpBuilder`, then use `super().machine` where `self.machine` is used.